### PR TITLE
JDK-8314426: runtime/os/TestTrimNative.java is failing on slow machines

### PR DESCRIPTION
**Pull request by tstuefe**
SAP reported errors on slow machines. Patch hardens the test against that.

- remove the grep for "Native Trimmer Stop" message, since that message is raced by the VM death and may not appear on slow machines
- for the low-interval test, made the number of trims to happen dependent on the runtime of the test. On slow machines, we could run longer than 3 seconds and therefore see more trim messages

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8314426](https://bugs.openjdk.org/browse/JDK-8314426): runtime/os/TestTrimNative.java is failing on slow machines (**Bug** - P4)

### Reviewers
 * [Matthias Baesken](https://openjdk.org/census#mbaesken) (@MBaesken - **Reviewer**) ⚠️ Review applies to [087687b4](https://git.openjdk.org/jdk/pull/15309/files/087687b46272c642d218b25b7cdefa98aa2a3e18)
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**) ⚠️ Review applies to [087687b4](https://git.openjdk.org/jdk/pull/15309/files/087687b46272c642d218b25b7cdefa98aa2a3e18)
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to [087687b4](https://git.openjdk.org/jdk/pull/15309/files/087687b46272c642d218b25b7cdefa98aa2a3e18)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15309/head:pull/15309` \
`$ git checkout pull/15309`

Update a local copy of the PR: \
`$ git checkout pull/15309` \
`$ git pull https://git.openjdk.org/jdk.git pull/15309/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15309`

View PR using the GUI difftool: \
`$ git pr show -t 15309`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15309.diff">https://git.openjdk.org/jdk/pull/15309.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15309#issuecomment-1687900363)

### CODE REVIEW

**Review by MBaesken — APPROVED** [review]

File: test/hotspot/jtreg/runtime/os/TestTrimNative.java [hunk 1]
```diff
@@ -296,12 +295,16 @@ public static void main(String[] args) throws Exception {
...
             case "trimNativeLowInterval":
             case "trimNativeLowIntervalStrict": {
+                long ms1 = System.currentTimeMillis();
                 OutputAnalyzer output = runTestWithOptions(
                         new String[] { "-XX:+UnlockExperimentalVMOptions", "-XX:TrimNativeHeapInterval=1" },
                         new String[] { TestTrimNative.Tester.class.getName(), "0" }
                 );
+                long ms2 = System.currentTimeMillis();
+                long runtime_ms = ms2 - ms1;
+                long maxTrimsExpected = runtime_ms;
```
Comment (TheRealMDoerr) [inline test/hotspot/jtreg/runtime/os/TestTrimNative.java:305]: Maybe add a comment explaining why you assign this.

**Review by TheRealMDoerr — APPROVED** [review]
Thanks for improving it!

File: test/hotspot/jtreg/runtime/os/TestTrimNative.java (hunk 1 above)
Comment (shipilev) [inline test/hotspot/jtreg/runtime/os/TestTrimNative.java:305]: Fixes the style and improves code a bit:

```suggestion
                int maxTrimsExpected = (int)(ms2 - ms1);
```

Then `(int)maxTrimsExpected` does not need a cast too.

**Review by shipilev — APPROVED** [review]
Looks fine, minor nits.

**Pull request closed without merging**